src/app/clusters/on-off-server/on-off-server.cpp

```diff
@@ -221,14 +227,18 @@ EmberAfStatus OnOffServer::setOnOffValue(chip::EndpointId endpoint, chip::Comman
 #endif
 #ifdef EMBER_AF_PLUGIN_MODE_SELECT
         // If OnMode is not a null value, then change the current mode to it.
-        if (emberAfContainsServer(endpoint, ModeSelect::Id) &&
-            emberAfContainsAttribute(endpoint, ModeSelect::Id, ModeSelect::Attributes::OnMode::Id))
+        for (const auto& element : ModeSelect::Instance::ModeSelectAliasesInstanceMap)
         {
-            ModeSelect::Attributes::OnMode::TypeInfo::Type onMode;
-            if (ModeSelect::Attributes::OnMode::Get(endpoint, onMode) == EMBER_ZCL_STATUS_SUCCESS && !onMode.IsNull())
+            auto modeSelectAlias = element.second;
+            if (modeSelectAlias->HasFeature(ModeSelect::ModeSelectFeature::kDeponoff))
```

Comment:
Where are we checking the endpoint here?

Reply:
[Fixed here](https://github.com/project-chip/connectedhomeip/pull/26447/commits/fe822bb99089aff69e79b20f73ac064114318f8e).